### PR TITLE
new DPU Gzipper

### LINKED ISSUE #210: T-GZipper: support for gzip compression
Create new DPU or adjust existing (t-zipper) with support for gnu zip compression to compress files into gzip form. Adjust also Readme.md properly.

### PR DESCRIPTION
**Pull request by mbl-eea-sk**
Resolve #210 new DPU Gzipper

### CODE REVIEW

File: t-gzipper/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+|**N/A**                                         |N/A                |
+
+***
+
+### Inputs and outputs ###
+
+|Name                |Type       |DataUnit                         |Description                        |
+|--------------------|-----------|---------------------------------|-----------------------------------|
+|input |i |FilesDataUnit  |One or more files |
+|output|o |FilesDataUnit  |One .gz file per each input file |
```
Comment (tomas-knap) [inline t-gzipper/README.md]: This is strange - I would expect that the output is just one file containing gzipped input files.

**Comment by tomas-knap** [comment]
Also the DPU, when executed, should send event saying "10/12 input files was zipped to xx.gz", skipped files Y, Z.

File: t-gzipper/src/test/java/eu/unifiedviews/plugins/transformer/gzipper/GzipperTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        dpu.configure((new ConfigurationBuilder()).setDpuConfiguration(config).toString());
+
+        // Prepare test environment.
+        TestEnvironment environment = new TestEnvironment();
+
+        // Prepare data unit.
+        WritableFilesDataUnit filesOutput = environment.createFilesOutput("filesOutput");
+        WritableFilesDataUnit filesInput = environment.createFilesInput("filesInput");
+
+        File inputFile = new File(URI.create(filesInput.addNewFile("LICENSE.gz")));
```
Comment (tomas-knap) [inline t-gzipper/src/test/java/eu/unifiedviews/plugins/transformer/gzipper/GzipperTest.java]: This is strange, input should be non-gzipped file

**Comment by tomas-knap** [comment]
Configuration must be similar to t-zipper, i.e. it must support definition of target file path/name:

screen shot 2015-06-26 at 12 57 20

File: t-gzipper/README.md (hunk 1 above)
Comment (skrchnavy) [inline t-gzipper/README.md]: gzip is not able to zip list of files just one.
We shall define if we shall make tar.gz or gz outputs (via configuration).

File: t-gzipper/src/test/java/eu/unifiedviews/plugins/transformer/gzipper/GzipperTest.java (hunk 2 above)
Comment (mbl-eea-sk) [inline t-gzipper/src/test/java/eu/unifiedviews/plugins/transformer/gzipper/GzipperTest.java]: It is a test input. As no compression makes two identical compressed files when running more times, this test unzips the compressed file and compares it to the original one.

File: t-gzipper/README.md (hunk 1 above)
Comment (mbl-eea-sk) [inline t-gzipper/README.md]: Should be specified if the required output is one .tar.gz file. Gzip compression does not support multiple files archive.

**Comment by mkl-eea-sk** [comment]
@tomas-knap you requirements is invalid:
http://www.gzip.org/#faq16

**Comment by mkl-eea-sk** [comment]
Ad configuration of output file name: this can be added, but would work for one file on input of course. 
If we would do the dialog for more than one file... well , that is work that should be done in t-filesRenamer, so lets do it there (mass renamer).

By default , DPU work on 'per file' basis, where we copy fileNames from input to output. I was talking to @hanecak giving hint that we should unify the way our DPUs work. For example, most users expect that gzip will make from aaa.xls: aaa.xls.gz, gunzip will take aaa.xls.gz and produce aaa.xls, if gunzip is given aaa.xls, it names the result aaa.xls (it does not know what to strip). Our DPUs do not behave this standard way - so, write a guideline or discuss with @hanecak what to do.

**Comment by skrchnavy** [comment]
This is moved to 2.2.0 version of plugins, so state moved to planed, after release of 2.1.0 this will be put back to review process.

**Comment by mkl-eea-sk** [comment]
Status: planned means for me> "start doing some work to resolve issue". from your comment, it seems that this is not the case, so move it elsewhere please :)

**Comment by skrchnavy** [comment]
Some work need to be done, need discussion what is expected.

**Comment by skrchnavy** [comment]
moved back to milestone 2.1.0. GZIP DPU produces gzip file per file on input.
Later can be extended to use tar (or another archive utility) to produce one file with all input files.
@tomas-knap please review

**Comment by tomas-knap** [comment]
Ok, if it gzips only one input to one output, than OK. 

Nevertheless, if the input file to be zipped is xxx, then I would expect that the output file is xxx.gz, as this is standard extension for gzipped files. Regarding ungzipper, it should try to unzip any file (whether it is named xxx or xxx.gz) and if xx.gz was used, the resulting unzipped file is xxx. So either this should be adjusted in this pull request or separate issue must be created which will solve that for the next release.

**Comment by mkl-eea-sk** [comment]
Gzipper adjusted - appends .gz. Ungzipper issue created: #253.
Comment https://github.com/UnifiedViews/Plugins/pull/239#issuecomment-117303728 fixed, feature finished.

**Pull request closed without merging**